moto/core/exceptions.py

```diff
@@ -178,3 +181,40 @@ class InvalidToken(AWSError):
 
     def __init__(self, message: str = "Invalid token"):
         super().__init__(f"Invalid Token: {message}", "InvalidToken")
+
+
+class MotoDockerException(RuntimeError):
```

Comment:
Having a custom exception is not a good solution IMO.
 - It couples the logic very closely to the tests
 - It doesn't work when testing ServerMode, where we only communicate with Moto over HTTP requests.

Reply:
"It couples the logic very closely to the tests" That's usually the case with unit tests, they are very clearbox oriented. 

As you may have noticed, these are not raised by default. This is more like a helper exception, so that the related tests will hard fail when something breaks that is not docker related, instead of softfailing. That exception is only used by the requires_docker marker/decorator. The tests themselves have no awareness.

From code perspective, it only knows about: in case of error raise, in addition to the current logic already in place to capture the error and providing some non-aws response. I only had to create some hoops for the run-in-separate-thread situation to have the exception keep bubbling up.

Alternative is to add various checks in the tests to see if some data returned contains some indication that docker is having issues, which I find really ugly and does couples the code and tests even closer (besides from the many additional non-dry code getting in the various tests).

It did help me uncover some corner cases that I did not detect before as likely/potentially impacted by not having docker running.

Most users will not use the xfails method, but it's a good one for sanity checks IMHO.

I'll have to run with servermode as well, and see how that works out / can be handled.

If you prefer, I can remove the exception related bits together with the xfail.

Reply:
instead of optionally raising this custom exception, we can make sure that when docker acts up, also a proper log message is generated.